Pidaq/Lib/database/database.py

```diff
@@ -0,0 +1,239 @@
+                   for current_database in current_databases):
+            self.client.create_database(database)
+        self.client.switch_database(database)
+        self.current_database = database
+
+    def switch_database(self, database: str) -> None:
+        """Switches currently active database"""
+
+        current_databases = self.client.get_list_database()
+        if not any(self.current_database['name'] == database
+                   for self.current_database in current_databases):
+            self.client.create_database(database)
+        self.client.switch_database(database)
+        self.current_database = database
+
+    def log_data(self, data: dict, measurement: str, tags: dict) -> None:
+        """Logs a single data point to the database"""
+
+        table_row = [{
+            'measurement': measurement,
+            'time': datetime.now(),
+            'fields': data,
+        }]
+        self.client.write_points(table_row, tags=tags)
+
+    def read_data(self, query=None, tags=None, fields=None, measurements=None) -> str:
+        """Reads database for specified measurements, including specified tags and fields
+
+        Format of a query when not using query parameter:
+            SELECT fields FROM measurements WHERE tags.keys = tags.values
+
+        Args:
+            query (str): if argument is passed, all other parameters are ignored, and database
+                         is queried
+            tags (dict): Contains the tag keys and values to be included in the query
+            fields (list(str)): Contains fields to be queried
+            measurements (list(str)): Contains the measurements to be queried from
+
+        Returns:
+            (str) Raw json data
+        """
+
+        if query is not None:
+            data = self.client.query(query, database=self.current_database)
+        elif measurements is None:
+            return -1
+        else:
+            formatted_measurements = ','.join([f'"{measurement}"' for measurement in measurements])
+            formatted_fields = ','.join([f'"{field}"' for field in fields]) if fields is not None else '*'
+            if tags is None:
+                data = self.client.query(
+                    f'SELECT {formatted_fields} FROM {formatted_measurements}',
+                    database=self.current_database
+                )
+            else:
+                formatted_tags = ' AND '.join(f""""{key}" = '{value}'""" for key, value in tags.items())
+                data = self.client.query(
+                    f'SELECT {formatted_fields} FROM {formatted_measurements} WHERE {formatted_tags}',
+                    database=self.current_database
+                )
+        try:
+            return data.raw['series']
+        except KeyError:
+            return -1
+
+    def create_retention_policy(self, name: str, duration: str, replication: int) -> None:
+        """Creates a retention policy for the current database"""
```

Comment:
Mention that retention policy duration is minimum 1 hr and include examples for duration